### PR TITLE
feat: Add infoButton slot to Field

### PR DESCRIPTION
**Pull request by behowell**
## Related Issue(s)

- Fixes #

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports                                                                                                |     Baseline (minified/GZIP) |                            PR |                                                                                                                                                                                                            Change |
| :--------------------------------------------------------------------------------------------------------------- | ---------------------------: | ----------------------------: | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: |
| <samp>react-checkbox</samp> <br /> <abbr title='bundle-size/CheckboxField.fixture.js'>CheckboxField</abbr>       | `39.996 kB`<br />`12.206 kB` | `153.501 kB`<br />`45.069 kB` | `113.505 kB` <img aria-hidden="true" src="https://microsoft.github.io/monosize/images/increase.png" /><br />`32.863 kB` <img aria-hidden="true" src="https://microsoft.github.io/monosize/images/increase.png" /> |
| <samp>react-combobox</samp> <br /> <abbr title='bundle-size/ComboboxField.fixture.js'>ComboboxField</abbr>       | `81.926 kB`<br />`27.213 kB` | `163.718 kB`<br />`48.818 kB` |  `81.792 kB` <img aria-hidden="true" src="https://microsoft.github.io/monosize/images/increase.png" /><br />`21.605 kB` <img aria-hidden="true" src="https://microsoft.github.io/monosize/images/increase.png" /> |
| <samp>react-field</samp> <br /> <abbr title='bundle-size/Field.fixture.js'>Field</abbr>                          |  `17.673 kB`<br />`6.543 kB` | `136.893 kB`<br />`41.366 kB` |  `119.22 kB` <img aria-hidden="true" src="https://microsoft.github.io/monosize/images/increase.png" /><br />`34.823 kB` <img aria-hidden="true" src="https://microsoft.github.io/monosize/images/increase.png" /> |
| <samp>react-input</samp> <br /> <abbr title='bundle-size/InputField.fixture.js'>InputField</abbr>                | `32.819 kB`<br />`10.111 kB` | `151.358 kB`<br />`44.342 kB` | `118.539 kB` <img aria-hidden="true" src="https://microsoft.github.io/monosize/images/increase.png" /><br />`34.231 kB` <img aria-hidden="true" src="https://microsoft.github.io/monosize/images/increase.png" /> |
| <samp>react-progress</samp> <br /> <abbr title='bundle-size/ProgressField.fixture.js'>ProgressField</abbr>       |  `23.885 kB`<br />`8.385 kB` |  `142.93 kB`<br />`42.924 kB` | `119.045 kB` <img aria-hidden="true" src="https://microsoft.github.io/monosize/images/increase.png" /><br />`34.539 kB` <img aria-hidden="true" src="https://microsoft.github.io/monosize/images/increase.png" /> |
| <samp>react-radio</samp> <br /> <abbr title='bundle-size/RadioGroupField.fixture.js'>RadioGroupField</abbr>      |   `24.963 kB`<br />`9.07 kB` | `138.805 kB`<br />`41.904 kB` | `113.842 kB` <img aria-hidden="true" src="https://microsoft.github.io/monosize/images/increase.png" /><br />`32.834 kB` <img aria-hidden="true" src="https://microsoft.github.io/monosize/images/increase.png" /> |
| <samp>react-select</samp> <br /> <abbr title='bundle-size/SelectField.fixture.js'>SelectField</abbr>             | `33.384 kB`<br />`10.779 kB` | `152.285 kB`<br />`45.126 kB` | `118.901 kB` <img aria-hidden="true" src="https://microsoft.github.io/monosize/images/increase.png" /><br />`34.347 kB` <img aria-hidden="true" src="https://microsoft.github.io/monosize/images/increase.png" /> |
| <samp>react-slider</samp> <br /> <abbr title='bundle-size/SliderField.fixture.js'>SliderField</abbr>             | `42.922 kB`<br />`13.465 kB` | `156.402 kB`<br />`46.072 kB` |  `113.48 kB` <img aria-hidden="true" src="https://microsoft.github.io/monosize/images/increase.png" /><br />`32.607 kB` <img aria-hidden="true" src="https://microsoft.github.io/monosize/images/increase.png" /> |
| <samp>react-spinbutton</samp> <br /> <abbr title='bundle-size/SpinButtonField.fixture.js'>SpinButtonField</abbr> | `41.766 kB`<br />`12.254 kB` | `160.539 kB`<br />`46.623 kB` | `118.773 kB` <img aria-hidden="true" src="https://microsoft.github.io/monosize/images/increase.png" /><br />`34.369 kB` <img aria-hidden="true" src="https://microsoft.github.io/monosize/images/increase.png" /> |
| <samp>react-switch</samp> <br /> <abbr title='bundle-size/SwitchField.fixture.js'>SwitchField</abbr>             |  `35.01 kB`<br />`10.655 kB` | `149.287 kB`<br />`43.646 kB` | `114.277 kB` <img aria-hidden="true" src="https://microsoft.github.io/monosize/images/increase.png" /><br />`32.991 kB` <img aria-hidden="true" src="https://microsoft.github.io/monosize/images/increase.png" /> |
| <samp>react-textarea</samp> <br /> <abbr title='bundle-size/TextareaField.fixture.js'>TextareaField</abbr>       | `36.537 kB`<br />`11.504 kB` | `155.074 kB`<br />`45.428 kB` | `118.537 kB` <img aria-hidden="true" src="https://microsoft.github.io/monosize/images/increase.png" /><br />`33.924 kB` <img aria-hidden="true" src="https://microsoft.github.io/monosize/images/increase.png" /> |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |          Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: |
| <samp>react-checkbox</samp> <br /> <abbr title='bundle-size/Checkbox.fixture.js'>Checkbox</abbr>                                                                              |  `33.091 kB`<br />`10.319 kB` |
| <samp>react-combobox</samp> <br /> <abbr title='bundle-size/Combobox.fixture.js'>Combobox (including child components)</abbr>                                                 |  `85.509 kB`<br />`27.437 kB` |
| <samp>react-combobox</samp> <br /> <abbr title='bundle-size/Dropdown.fixture.js'>Dropdown (including child components)</abbr>                                                 |  `84.196 kB`<br />`27.191 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr>              |   `64.09 kB`<br />`17.544 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `202.311 kB`<br />`56.366 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            |  `35.049 kB`<br />`11.533 kB` |
| <samp>react-input</samp> <br /> <abbr title='bundle-size/Input.fixture.js'>Input</abbr>                                                                                       |    `22.94 kB`<br />`7.214 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr>                                                         |    `6.324 kB`<br />`2.129 kB` |
| <samp>react-progress</samp> <br /> <abbr title='bundle-size/ProgressBar.fixture.js'>ProgressBar</abbr>                                                                        |     `13.59 kB`<br />`5.24 kB` |
| <samp>react-radio</samp> <br /> <abbr title='bundle-size/Radio.fixture.js'>Radio</abbr>                                                                                       |  `32.676 kB`<br />`10.616 kB` |
| <samp>react-radio</samp> <br /> <abbr title='bundle-size/RadioGroup.fixture.js'>RadioGroup</abbr>                                                                             |   `14.885 kB`<br />`5.933 kB` |
| <samp>react-select</samp> <br /> <abbr title='bundle-size/Select.fixture.js'>Select</abbr>                                                                                    |   `24.145 kB`<br />`8.271 kB` |
| <samp>react-slider</samp> <br /> <abbr title='bundle-size/Slider.fixture.js'>Slider</abbr>                                                                                    |  `33.091 kB`<br />`10.555 kB` |
| <samp>react-spinbutton</samp> <br /> <abbr title='bundle-size/SpinButton.fixture.js'>SpinButton</abbr>                                                                        |    `32.83 kB`<br />`9.864 kB` |
| <samp>react-switch</samp> <br /> <abbr title='bundle-size/Switch.fixture.js'>Switch</abbr>                                                                                    |   `28.225 kB`<br />`8.755 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/DataGrid.fixture.js'>DataGrid</abbr>                                                                                 | `144.785 kB`<br />`39.447 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/PrimitivesOnly.fixture.js'>Table (Primitives only)</abbr>                                                            |  `43.776 kB`<br />`12.063 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/TableAsDataGrid.fixture.js'>Table as DataGrid</abbr>                                                                 |   `134.4 kB`<br />`34.514 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/TableSelectionOnly.fixture.js'>Table (Selection only)</abbr>                                                         |  `82.316 kB`<br />`20.576 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/TableSortOnly.fixture.js'>Table (Sort only)</abbr>                                                                   |  `81.637 kB`<br />`20.386 kB` |
| <samp>react-textarea</samp> <br /> <abbr title='bundle-size/Textarea.fixture.js'>Textarea</abbr>                                                                              |   `26.336 kB`<br />`8.585 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/4464df2ab9683e09bb456f60bdb5e79199b9c5d3'>4464df2ab9683e09bb456f60bdb5e79199b9c5d3</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27032/perf-test-react-components/Avatar-mount_base.html">1324</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27032/perf-test-react-components/Avatar-mount.html">1316</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27032/perf-test-react-components/Button-mount_base.html">960</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27032/perf-test-react-components/Button-mount.html">952</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27032/perf-test-react-components/Field-mount_base.html">2063</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27032/perf-test-react-components/Field-mount.html">2080</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27032/perf-test-react-components/FluentProvider-mount_base.html">1611</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27032/perf-test-react-components/FluentProvider-mount.html">1656</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27032/perf-test-react-components/FluentProviderWithTheme-mount_base.html">641</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27032/perf-test-react-components/FluentProviderWithTheme-mount.html">632</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27032/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">599</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27032/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">602</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27032/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">630</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27032/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">625</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>InfoButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27032/perf-test-react-components/InfoButton-mount_base.html">570</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27032/perf-test-react-components/InfoButton-mount.html">573</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27032/perf-test-react-components/MakeStyles-mount_base.html">1896</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27032/perf-test-react-components/MakeStyles-mount.html">1914</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27032/perf-test-react-components/Persona-mount_base.html">3036</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27032/perf-test-react-components/Persona-mount.html">3012</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27032/perf-test-react-components/SpinButton-mount_base.html">2474</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27032/perf-test-react-components/SpinButton-mount.html">2439</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

File: apps/vr-tests-react-components/src/stories/Field.stories.tsx [hunk 1]
```diff
@@ -109,6 +109,44 @@ storiesOf('Field', module)
       <Checkbox label="Checkbox in a horizontal field" />
```
Comment (fabricteam) [inline apps/vr-tests-react-components/src/stories/Field.stories.tsx:109]: ## [🕵 fluentuiv9 Open the Visual Regressions report to inspect the 7 screenshots]( https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-fluent/index.html?organization=uifabric&project=fabricpublic&prId=27032&commitId=af8e089011664c9f7cf1f72e96717024938e9ba7&env=prod)

✅ There was 7 screenshots added, 0 screenshots removed, 1862 screenshots unchanged, 0 screenshots with different dimensions and 0 screenshots with visible difference.


  <details><summary><b>unknown</b> 7 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Field.infoButton+horizontal.default.chromium.png|0| Added|
|Field.infoButton+longLabel.default.chromium.png|0| Added|
|Field.infoButton+noLabel.default.chromium.png|0| Added|
|Field.infoButton+required.default.chromium.png|0| Added|
|Field.infoButton+size-large.default.chromium.png|0| Added|
|Field.infoButton+size-small.default.chromium.png|0| Added|
|Field.infoButton.default.chromium.png|0| Added|

</details>


<div id="vrtCommentfluentuiv9"/>

**Comment by behowell** [comment]
Abandoning in favor of the approach used in #27030, due to the bundle size cost of referencing the InfoButton component directly from Field.

**Pull request closed without merging**